lua/neoclip/db.lua

```diff
@@ -0,0 +1,70 @@
+local M = {}
+
+local has_sqlite, sqlite = pcall(require, "sqlite")
+if not has_sqlite then
+    print "Couldn't find sqlite.lua. Cannot use persistent history"
+    return nil
+end
+
+local function get_db(db_path)
+    local db = sqlite.new(db_path)
+    db:open()
+    db:create('neoclip', {
+        id = true,
+        ensure = true,
+        regtype = 'text',
+        contents = 'text',
+        filetype = 'text',
+    })
+    return db
```

Comment:
`sqlite.extend` or `sqlite({...})` is encourage in creating db module, So it might be better to set `M` to  `sqlite({})`. 

- see [frecency db module](https://github.com/nvim-telescope/telescope-frecency.nvim/blob/refactor/lua/frecency/db.lua#L8-L25)

- see [`:h sqlite.db:extend`](https://github.com/tami5/sqlite.lua/blob/master/doc/sqlite.txt#L118-L139)

For `db_path` you can have something like M.set_config to set user config at startup/neocilp require and mutate `M.db.uri`. 

- see  [frecency.db.set_config](https://github.com/nvim-telescope/telescope-frecency.nvim/blob/refactor/lua/frecency/db.lua#L41-L46)

Reply:
Or if you really want to simplify it further set M to `sqlite.tbl("neoclip", { ... }, sqlite.new())`, then in `M.set_config` set path to `M.db.uri = db_path`, and when you overwrite for example `tbl:get` you can use `tbl:__get`.

Reply:
Thanks a lot for the references, I'll look into this :+1: